builtin/logical/pki/path_fetch_keys.go

```diff
@@ -0,0 +1,225 @@
+package pki
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+func pathListKeys(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "keys/?$",
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ListOperation: &framework.PathOperation{
+				Callback:                    b.pathListKeysHandler,
+				ForwardPerformanceStandby:   false,
+				ForwardPerformanceSecondary: false,
+			},
+		},
+
+		HelpSynopsis:    pathListKeysHelpSyn,
+		HelpDescription: pathListKeysHelpDesc,
+	}
+}
+
+const (
+	pathListKeysHelpSyn  = `Fetch a list of all issuer keys`
+	pathListKeysHelpDesc = `This endpoint allows listing of known backing keys, returning
+their identifier and their name (if set).`
+)
+
+func (b *backend) pathListKeysHandler(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	var responseKeys []string
+	responseInfo := make(map[string]interface{})
+
+	entries, err := listKeys(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, identifier := range entries {
+		key, err := fetchKeyById(ctx, req.Storage, identifier)
+		if err != nil {
+			return nil, err
+		}
+
+		responseKeys = append(responseKeys, string(identifier))
+		responseInfo[string(identifier)] = map[string]interface{}{
+			keyNameParam: key.Name,
+		}
+
+	}
+	return logical.ListResponseWithInfo(responseKeys, responseInfo), nil
+}
+
+func pathKey(b *backend) *framework.Path {
+	pattern := "key/" + framework.GenericNameRegex(keyRefParam)
+	return buildPathKey(b, pattern)
+}
+
+func buildPathKey(b *backend, pattern string) *framework.Path {
+	return &framework.Path{
+		Pattern: pattern,
+
+		Fields: map[string]*framework.FieldSchema{
+			keyRefParam: {
+				Type:        framework.TypeString,
+				Description: `Reference to key; either "default" for the configured default key, an identifier of a key, or the name assigned to the key.`,
+				Default:     "default",
```

Comment:
nit: Please replace "default" with the constant defaultRef

Reply:
:) Done.